### PR TITLE
Optimize orthonormal basis generation using Pixar 2017 method

### PR DESCRIPTION
**Pull request by RohithPariki**
Resolves upstream issue microsoft/DirectXMath#73.

Implemented `XMVector3OrthogonalBasis` utilizing the Pixar 2017 method ("Building an Orthonormal Basis, Revisited" by Duff et al.) to compute a tangent and bitangent from a unit normal. This ensures numeric stability across all edge cases (particularly along the Z-axis, which causes singularities in naive cross-product derivations) while remaining branchless.

## changes:
- Added `XMVector3OrthogonalBasis` to `DirectXMath.h`
- Implemented `XMVector3OrthogonalBasis` in `DirectXMathVector.inl`
- Maintained scalar compatibility with `#if defined(_XM_NO_INTRINSICS_)`
- Optimized SIMD intrinsic execution pathway (reusing vectors and leveraging `XMVectorAndInt`/`XMVectorOrInt` for fast `copysign` replacement).

## how i verified.
I verified the implementation by writing a test script against the critical edge cases (vectors perfectly aligned with the axes, which often cause singularities in branchless ONB generation). 

```cpp
#include <iostream>
#include <DirectXMath.h>
using namespace DirectX;

void VerifyONB(float x, float y, float z) {
    XMVECTOR N = XMVectorSet(x, y, z, 0);
    XMVECTOR T, B;
    
    // Generate Orthogonal Basis
    XMVector3OrthogonalBasis(&T, &B, N);
    
    // Verify Orthogonality
    float dot_NT = XMVectorGetX(XMVector3Dot(N, T));
    float dot_NB = XMVectorGetX(XMVector3Dot(N, B));
    float dot_TB = XMVectorGetX(XMVector3Dot(T, B));
    
    XMFLOAT4 nf, tf, bf;
    XMStoreFloat4(&nf, N); XMStoreFloat4(&tf, T); XMStoreFloat4(&bf, B);
    
    std::cout << "Normal: (" << nf.x << ", " << nf.y << ", " << nf.z << ")\n";
    std::cout << "Tangent: (" << tf.x << ", " << tf.y << ", " << tf.z << ")\n";
    std::cout << "Bitangent: (" << bf.x << ", " << bf.y << ", " << bf.z << ")\n";
    std::cout << "Verification [N.T, N.B, T.B]: " << dot_NT << ", " << dot_NB << ", " << dot_TB << "\n\n";
}

int main() {
    VerifyONB(0, 0, 1);
    VerifyONB(0, 0, -1);
    VerifyONB(1, 0, 0);
    VerifyONB(0.57735f, 0.57735f, 0.57735f); // Arbitrary unit vector
    return 0;
}
```
the output :
```
Normal: (0, 0, 1)
Tangent: (1, -0, -0)
Bitangent: (-0, 1, -0)
Verification [N.T, N.B, T.B]: 0, 0, 0

Normal: (0, 0, -1)
Tangent: (1, -0, 0)
Bitangent: (0, -1, -0)
Verification [N.T, N.B, T.B]: 0, 0, 0

Normal: (1, 0, 0)
Tangent: (0, -0, -1)
Bitangent: (-0, 1, -0)
Verification [N.T, N.B, T.B]: 0, 0, 0

Normal: (0.57735, 0.57735, 0.57735)
Tangent: (0.788675, -0.211325, -0.57735)
Bitangent: (-0.211325, 0.788675, -0.57735)
Verification [N.T, N.B, T.B]: 3.35524e-07, 3.35524e-07, -1.10278e-07
```

### CODE REVIEW

File: Inc/DirectXMathVector.inl [hunk 1]
```diff
@@ -10216,6 +10216,87 @@ inline XMVECTOR XM_CALLCONV XMVector3Orthogonal(FXMVECTOR V) noexcept
...
+    // b2.z = -n.y
+    XMVECTOR B2Z = XMVectorNegate(Y);
+
+    // Combine into tangent and bitangent
+    XMVECTOR Zero = XMVectorZero();
+
+    // pTangent = (B1X, B1Y, B1Z, 0)
+    XMVECTOR B1_XY = XMVectorMergeXY(B1X, B1Y);
+    XMVECTOR B1_Z0 = XMVectorMergeXY(B1Z, Zero);
+    *pTangent = XMVectorPermute<XM_PERMUTE_0X, XM_PERMUTE_0Y, XM_PERMUTE_1X, XM_PERMUTE_1Y>(B1_XY, B1_Z0);
```
Comment (walbourn) [inline Inc/DirectXMathVector.inl]: Should this be `if (pTangent) { *pTangen t==...}` so the parameter can be `_Out_opt_`?

File: Inc/DirectXMathVector.inl [hunk 2]
```diff
@@ -10216,6 +10216,87 @@ inline XMVECTOR XM_CALLCONV XMVector3Orthogonal(FXMVECTOR V) noexcept
...
+    _Out_ XMVECTOR* pTangent,
+    _Out_ XMVECTOR* pBitangent,
+    _In_ FXMVECTOR Normal
+) noexcept
+{
+    assert(pTangent != nullptr);
+    assert(pBitangent != nullptr);
+
+#if defined(_XM_NO_INTRINSICS_)
+    float sign = copysignf(1.0f, Normal.vector4_f32[2]);
```
Comment (walbourn) [inline Inc/DirectXMathVector.inl]: Might have to investigate how far back `copysignf` can be found in MSVC. There are some compilation scenarios for DirectXMath that may require a different solution.
